src/fileimpl.cpp

```diff
@@ -417,7 +417,26 @@ makeFileReader(std::shared_ptr<const FileCompound> zimFile, offset_t offset, zsi
     if (idx >= getCountClusters())
       throw ZimFileFormatError("cluster index out of range");
 
-    return clusterCache.getOrPut(idx.v, [=](){ return readCluster(idx); });
+    auto cluster = clusterCache.getOrPut(idx.v, [=](){ return readCluster(idx); });
+#if ENV32BIT
+    // There was a bug in the way we create the zim files using ZSTD compression.
+    // We were using a too hight compression level and so a window of 128Mb.
+    // So at decompression, zstd reserve a 128Mb buffer.
+    // While this memory is not really used (thanks to lazy allocation of OS),
+    // we are still consumming address space. On 32bits this start to be a rare
+    // ressource when we reserved 128Mb at once.
+    // So we drop the cluster from the cache to avoid future memory allocation error.
+    if (cluster->getCompression() == zimcompZstd) {
+      // ZSTD compression starts to be used on version 5.0 of zim format.
+      // Recently after, we switch to 5.1 and itegrate the fix in zstd creation.
+      // 5.0 is not a perfect way to detect faulty zim file (it will generate false
+      // positives) but it should be enough.
+      if (header.getMajorVersion() == 5 && header.getMinorVersion() == 0) {
+        clusterCache.drop(idx.v);
+      }
```

Comment:
This may lead to very significant slowdown because `Item::getSize()` and `Item::getData()` become prohibitively expensive.

Reply:
Yes, this is the downside of the workaround.
But it will happen only :
- On 32bits. Mainly windows command line tools (desktop is 64bit) and armhf (and few android/ios devices)
- For only zstd zim made with version 6.3.* (so we need to release quickly version 7)

The other potential workaround is to resurrect the old code without partial decompression and use it instead of dropping the cluster from the cache.
But then we would have a double implementation we would have to maintain over time and I prefer to avoid that.
If we release libzim7, this workaround will be use less and less quickly as we generate new zim files.